### PR TITLE
partitioned-table: Correct the variable name for list partitioning

### PR DESCRIPTION
**Pull request by dveeden**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

List partitioning requires the `tidb_enable_list_partition` variable to be set to ON, not just `tidb_enable_table_partition`. Updating the text for this.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: partitioned-table.md [hunk 1]
```diff
@@ -170,14 +170,16 @@ Range partitioning is particularly useful when one or more of the following cond
...
+Before creating a List partitioned table, you need to set the value of the session variable `tidb_enable_list_partition` to `ON`.
 
 {{< copyable "sql" >}}
 
 ```sql
-set @@session.tidb_enable_table_partition = ON
+set @@session.tidb_enable_list_partition = ON
 ```
 
+Also `tidb_enable_table_partition` needs to be set to `ON`, but this is the default.
```
Comment (TomShawn) [inline partitioned-table.md:181]:
```suggestion
Also, make sure that `tidb_enable_table_partition` is set to `ON`, which is the default setting.
```

Comment (crazycs520) [reply partitioned-table.md:181]: Please address this comment.

REST LGTM

**Pull request merged**